src/lexer.rs

```diff
@@ -31,6 +31,11 @@ impl Lexer {
                 result.push(self.lex_string());
             }
         }
+        result.push(Token {
+            value: EOF.to_string(),
+            token_type: TokenType::Eof,
+            pos: EOF_POS,
```

Comment:
What about letting the position be the position of the '\0', that is, src.len()?
The space is taken anyway for every EOF token, so it might be a little beneficial to have the position representing some info.